mapmetadata.c

```diff
@@ -687,6 +687,11 @@ xmlNodePtr msMetadataGetLayerMetadata(mapObj *map, metadataParamsObj *paramsObj,
     if(strcasecmp(GET_LAYER(map, i)->name, paramsObj->pszLayer) == 0) {
         layer_found = MS_TRUE;
         layer = GET_LAYER(map, i);
+        // when checking a layer with clustering msLayerGetExtent does not have access
```

Comment:
I have the feeling this isn't the right fix. That wouldn't fix the crash for someone calling msLayerGetExtent() through the C API or Mapscript, would it ?

Reply:
This function is only called using the `GetMetadata` request (see https://mapserver.org/ogc/layer_metadata.html) - there is no mapscript access currently.

Reply:
> This function is only called using the `GetMetadata` request

yes, but isn't the issue fundamentally with msLayerGetExtent() called on a clustered layer ?

Reply:
True..  Cluster layers are a special case in [msLayerOpen](https://github.com/MapServer/MapServer/blob/4505f7028b20f8d86734419450c2f3299bf76ace/mapcluster.c#L1622). Then a copy is made of the original layer in [msClusterLayerOpen](https://github.com/MapServer/MapServer/blob/4505f7028b20f8d86734419450c2f3299bf76ace/mapcluster.c#L1622), but it is the source layer vtable which is opened, but then the copied layer doesn't have the original connection - I had a hard time trying to follow exactly what is happening here. 
I've not had any cases of `msLayerGetExtent` calls crashing the server apart from through `GetMetadata`, but this may be luck/missing logs. The fix in this pull request prevents a server crash, but I agree it would be better to fix the underlying issue.